### DISCUSSION TITLE
Which node version are you using for contributing? v16 changes all minified files.

### DISCUSSION THREAD

**Discussion by ghost** [PrismJS/prism#2898] [Q&A] [answered]
I tried node@10, node@12, node@14, node@16 but every version changes all minified files.

Which version are you guys using? I really hope if author writes `engine` into package.json.

**Comment by RunDevelopment** [comment, marked as answer]
You can use any [NodeJS version >=10.x](https://github.com/PrismJS/prism#software-requirements) to contribute.

> every version changes all minified files

You have to use `npm ci` to install Prism's dependencies. `npm install` is non-deterministic which causes non-deterministic builds, hence the minified files changed.

> `engine` into package.json

AFAIK, `engine` specifies which NodeJS version is necessary to use Prism as a library and not which version is necessary to develop Prism.

**Reply by ghost** [reply]
@RunDevelopment ohh, thank you. I didn't do that, and now it fixed. Thank you very much.

> AFAIK, engine specifies which NodeJS version is necessary to use Prism as a library and not which version is necessary to develop Prism.

Yes, now I know it's my misunderstanding, but when I faced all minified files changed, I just tried all versions in vain. So I wanted to add some informations somewhere, not only package.json but also ndenv or asdf.

now it's solved, thank you.
